### PR TITLE
Add 'generate-etc' django-admin command

### PR DESCRIPTION
**Pull request by jkokosar**

### CODE REVIEW

File: resolwe_bio/management/commands/generate_etc.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    help = "Generate ETC objects"
+
+    def add_arguments(self, parser):
+        parser.add_argument('-e', '--n-etc', type=int, default=3,
+                            help="Number of ETC objects to generate (default: %(default)s)")
+        parser.add_argument('--rseed', action='store_true', help="Use fixed random seed")
+
+
+    def set_user(self):
+        users = User.objects.filter(is_superuser=True).order_by('date_joined')
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Use `from django.contrib.auth import get_user_model` and then `get_user_model().objects...`.

This will work even if other user model is used.

File: resolwe_bio/descriptors/dicty.yml [hunk 2]
```diff
@@ -178,8 +178,8 @@
               value: DIF treatment
             - label: Percoll gradient (Prespore cells - Heavy fraction)
               value: Percoll gradient (Prespore cells - Heavy fraction)
-            - label: Percoll gradient (Prestalk cells - Light fraction) 
-              value: Percoll gradient (Prestalk cells - Light fraction) 
+            - label: Percoll gradient (Prestalk cells - Light fraction)
```
Comment (dblenkus) [inline resolwe_bio/descriptors/dicty.yml]: Are these changes related to the new command? If not, put them in separate commit.

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -46,6 +46,7 @@ Added
 * API endpoint for adding ``Samples`` to ``Collections``
 * HISAT2 aligner
 * Use Git Large File Storage (LFS) for large test files
+*	django-admin command: ``generate_etc``
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Space instead of tab

File: resolwe_bio/management/commands/generate_etc.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                for gene in all_genes:
+                    expression = [str(random.gammavariate(1, 100)) for i in range(7)]
+                    expression = '\t'.join(expression)
+                    f.write('{}\t{}\n'.format(gene, expression))
+
+        with open(os.path.join(path, file_name)) as csvfile:
+            my_reader = csv.reader(csvfile, delimiter=str('\t'))
+            times = [int(x) for x in next(my_reader)[1:]]
+            for x in my_reader:
+                genes[escape_mongokey(str(x[0]))] = x[1:]
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Is `escape_mongokey` still needed as we are not using mongo anymore? Iy yes, consider changing name.

File: resolwe_bio/management/commands/generate_etc.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        with open(os.path.join(path, file_name), 'w') as f:
+            f.write('\t'.join(['Gene', '0', '4', '8', '12', '16', '20', '24']) + '\n')
+            with gzopen(gene_ids) as gene_ids:
+                all_genes = [line.strip() for line in gene_ids]
+                for gene in all_genes:
+                    expression = [str(random.gammavariate(1, 100)) for i in range(7)]
+                    expression = '\t'.join(expression)
+                    f.write('{}\t{}\n'.format(gene, expression))
+
+        with open(os.path.join(path, file_name)) as csvfile:
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Don't read newly created file. Generate data in memory and then write them in two files.

File: resolwe_bio/management/commands/generate_etc.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+    help = "Generate ETC objects"
+
+    def add_arguments(self, parser):
+        parser.add_argument('-e', '--n-etc', type=int, default=3,
+                            help="Number of ETC objects to generate (default: %(default)s)")
+        parser.add_argument('--rseed', action='store_true', help="Use fixed random seed")
+
+
+    def set_user(self):
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: You don't actually set the user in this function. Rename it to `get_user` or even better `get_superuser`.

**Comment by dblenkus** [comment]
@jkokosar is there a reason why you don't use `csv.writer` for creating CSV files?

File: resolwe_bio/management/commands/generate_etc.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        dicty_genes = os.path.join(test_files_path, 'dicty_genes.tab.gz')
+
+        # Create reads data object
+        started = timezone.now()
+        etc = Data.objects.create(
+            slug='etc',
+            name="D. discoideum",
+            started=started,
+            finished=started + datetime.timedelta(minutes=1),
+            descriptor_schema=DescriptorSchema.objects.get(slug='dicty-etc'),
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Check if DescriptorSchema exists before assigning it and raise meaningful if not.

File: resolwe_bio/management/commands/generate_etc.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+            descriptor=self.generate_etc_desciptor(),
+            status=Data.STATUS_PROCESSING,
+            process=Process.objects.get(slug='upload-etc'),
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Also check for `Process`

File: resolwe_bio/management/commands/generate_etc.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+            self.stderr.write("Process ({}) does not exist: try running './manage.py register' command.".format(slug))
+            exit(1)
+        return process
+
+
+    @staticmethod
+    def create_etc(gene_ids, path, file_name):
+        genes = {}
+        with open(os.path.join(path, file_name), 'w') as f:
+            csvwriter = csv.writer(f, delimiter=str('\t'))
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Is `str` needed?

File: resolwe_bio/management/commands/generate_etc.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+        except DescriptorSchema.DoesNotExist:
+            self.stderr.write("Descriptor schema ({}) does not exist: "
+                "try running './manage.py register' command".format(slug))
+            exit(1)
+        return ds
+
+
+    def get_process(self, slug):
+        try:
+            process = Process.objects.get(slug=slug)
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: This will raise error if there are multiple `Processes` with same `slug`.

```
process_qs = Process.objects.filter(slug=slug)
if not process_qs.exists():
    self.stderr.write("Process ({}) does not exist: try running './manage.py register' command.".format(slug))
    exit(1)
return process_qs.order_by('version').last()
```

File: resolwe_bio/management/commands/generate_etc.py [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+
+    def get_superuser(self):
+        users = get_user_model().objects.filter(is_superuser=True).order_by('date_joined')
+        if not users.exists():
+            self.stderr.write("Admin does not exist: create a superuser")
+            exit(1)
+        return users.first()
+
+    def get_descriptor_schema(self, slug):
```
Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Aame as for `get_process` function.

File: resolwe_bio/descriptors/dicty.yml (hunk 2 above)
Comment (dblenkus) [inline resolwe_bio/descriptors/dicty.yml]: @jkokosar?

**Comment by dblenkus** [comment]
Please rebase on #85 and start using `utils.py`

**Comment by jkokosar** [comment]
Rebased on https://github.com/genialis/resolwe-bio/pull/85

File: resolwe_bio/descriptors/dicty.yml (hunk 2 above)
Comment (jkokosar) [inline resolwe_bio/descriptors/dicty.yml]: Added to a separate commit. Had to learn about patch commits :)

File: resolwe_bio/management/commands/generate_etc.py (hunk 9 above)
Comment (jkokosar) [inline resolwe_bio/management/commands/generate_etc.py]: Yes, otherwise it fails with:
```csvwriter = csv.writer(f, delimiter='\t')
TypeError: "delimiter" must be string, not unicode```

Comment (dblenkus) [inline resolwe_bio/management/commands/generate_etc.py]: Ok, this is probably because of `from __future__ import unicode_literals`, but we have to keep it.

**Comment by dblenkus** [comment]
Rebased and merged
